packages/utils/src/utils/networked-storage.ts

```diff
@@ -222,38 +237,49 @@ export const getCidBlockVerifierFunction = (cid: CID, hasher: MultihashHasher):
  * Race block providers cancelling any pending requests once the block has been
  * found.
  */
-async function raceBlockRetrievers (cid: CID, providers: BlockRetriever[], hasher: MultihashHasher, options: AbortOptions & LoggerOptions): Promise<Uint8Array> {
+async function raceBlockRetrievers (cid: CID, blockBrokers: BlockBroker[], hasher: MultihashHasher, options: AbortOptions & LoggerOptions): Promise<Uint8Array> {
   const validateFn = getCidBlockVerifierFunction(cid, hasher)
 
   const controller = new AbortController()
   const signal = anySignal([controller.signal, options.signal])
 
+  const retrievers: Array<Required<Pick<BlockBroker, 'retrieve'>>> = []
+
+  for (const broker of blockBrokers) {
+    if (broker.retrieve != null) {
+      // @ts-expect-error retrieve may be undefined even though we've just
+      // checked that it isn't
+      retrievers.push(broker)
+    }
+  }
```

Comment:
an `isRetrievingBlockBroker` type guard function would be safer:

```suggestion
  for (const broker of blockBrokers) {
    if (isRetrievingBlockBroker(broker)) {
      retrievers.push(broker)
    }
  }
```